localstack-core/localstack/utils/bootstrap.py

```diff
@@ -521,17 +522,29 @@ def config_env_vars(cfg: ContainerConfiguration):
                     and not env_var.startswith("LOCALSTACK_")
                     and env_var not in profile_env
                 ):
-                    # Show a warning here in case we are directly forwarding an environment variable from
-                    # the system env to the container which has not been prefixed with LOCALSTACK_.
-                    # Suppress the warning for the "CI" env var.
-                    # Suppress the warning if the env var was set from the profile.
-                    LOG.warning(
-                        "Non-prefixed environment variable %(env_var)s is forwarded to the LocalStack container! "
-                        "Please use `LOCALSTACK_%(env_var)s` instead of %(env_var)s to explicitly mark this environment variable to be forwarded form the CLI to the LocalStack Runtime.",
-                        {"env_var": env_var},
-                    )
+                    # Collect all env vars that are directly forwarded from the system env
+                    # to the container which has not been prefixed with LOCALSTACK_ here.
+                    # Suppress the "CI" env var.
+                    # Suppress if the env var was set from the profile.
+                    non_prefixed_env_vars.append(env_var)
                 cfg.env_vars[env_var] = value
 
+        # collectively log deprecation warnings for non-prefixed sys env vars
+        if non_prefixed_env_vars:
+            from localstack.utils.analytics import log
+
+            for non_prefixed_env_var in non_prefixed_env_vars:
+                # Show a deprecation warning for each individual env var collected above
+                LOG.warning(
+                    "Non-prefixed environment variable %(env_var)s is forwarded to the LocalStack container! "
+                    "Please use `LOCALSTACK_%(env_var)s` instead of %(env_var)s to explicitly mark this environment variable to be forwarded form the CLI to the LocalStack Runtime.",
+                    {"env_var": non_prefixed_env_var},
+                )
+
+            log.event(
+                event="non_prefixed_cli_env_vars", payload={"env_vars": non_prefixed_env_vars}
+            )
```

Comment:
thought: somewhat unrelated but IMO we should start also writing tests for the analytics events to make sure they work as expected :thinking:

Reply:
Definitely, I thought about adding this, but then I postponed this to a follow-up to introduce unit tests for multiple log events with some utilities around it (mocking / asserting).